tests/ui/collect.stderr

```diff
@@ -0,0 +1,36 @@
+error: you are creating a collection of `result`s
+ --> $DIR/collect.rs:8:21
+  |
+8 |     let option_vec: Vec<_> = b.into_iter()
+  |                     ^^^^^^
+  |
+  = note: `-D possible-shortcircuiting-collect` implied by `-D warnings`
+help: if you are only interested in the `Ok` values or the first `Err`, try
+  |
+8 |     let option_vec: Result<std::vec::Vec<_>, _> = b.into_iter()
+  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: you are creating a collection of `option`s
+  --> $DIR/collect.rs:26:18
+   |
+26 |     let _result: Vec<_> = a.iter().map(Some).collect();
+   |                  ^^^^^^
+   |
+help: if you are only interested in the `Some` values or the first `None`, try
```

Comment:
The message about the first None seems a little weird

Reply:
Agreed. Fixed.